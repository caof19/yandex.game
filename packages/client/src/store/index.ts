import {
    useDispatch as useDispatchBase,
    useSelector as useSelectorBase,
    TypedUseSelectorHook,
    useStore as useStoreBase,
} from "react-redux";
import { combineReducers } from "redux";
import { configureStore } from "@reduxjs/toolkit";
import authSlice from "./slice/auth";
import themeSlice from "./slice/theme";

declare global {
    interface Window {
        APP_INITIAL_STATE: RootState;
    }
}

export const reducer = combineReducers({ auth: authSlice, theme: themeSlice });

export const store = configureStore({
    reducer,
    preloadedState:
        typeof window === "undefined" ? undefined : window.APP_INITIAL_STATE,
});

export type RootState = ReturnType<typeof reducer>;
export type AppDispatch = typeof store.dispatch;

export const useDispatch: () => AppDispatch = useDispatchBase;
export const useSelector: TypedUseSelectorHook<RootState> = useSelectorBase;
export const useStore: () => typeof store = useStoreBase;
