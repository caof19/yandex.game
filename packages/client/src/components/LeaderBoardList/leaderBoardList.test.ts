import { render } from "@testing-library/react";
import { LeaderBoardList } from ".";
import { store } from "../../store";
import { Provider } from "react-redux";
const leaderBoardTestData = [
    {
        data: {
            position: 1,
            username: "user001",
            peachFillerScore: 1000,
        },
    },
];
describe("Лидерборд", () => {
    it("Лидерборд рендерится без ошибок", () => {
        const { container } = render(
            LeaderBoardList({ list: leaderBoardTestData }),
            {
                wrapper: ({ children }) => Provider({ store: store, children }),
            },
        );
        expect(container).toMatchSnapshot();
    });

    it("Тестовые данные отрисовываются в компоненте лидерборда", () => {
        const { getByText } = render(
            LeaderBoardList({ list: leaderBoardTestData }),
        );
        expect(getByText("user001")).toBeTruthy();
    });
});
