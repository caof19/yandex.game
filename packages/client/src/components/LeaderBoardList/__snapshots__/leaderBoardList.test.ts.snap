// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Лидерборд Лидерборд рендерится без ошибок 1`] = `
<div>
  <div
    class="ant-list ant-list-split css-dev-only-do-not-override-1uq9j6g"
  >
    <div
      class="ant-spin-nested-loading css-dev-only-do-not-override-1uq9j6g"
    >
      <div
        class="ant-spin-container"
      >
        <li
          class="ant-list-item"
        >
          <div
            class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-align-center"
            style="gap: 32px;"
          >
            <div
              class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-align-center"
              style="gap: 16px;"
            >
              <span
                class="ant-typography css-dev-only-do-not-override-1uq9j6g"
              >
                <strong>
                  1
                </strong>
              </span>
              <span
                class="ant-typography css-dev-only-do-not-override-1uq9j6g"
              >
                <strong>
                  user001
                </strong>
              </span>
            </div>
            <span
              class="ant-typography ant-typography-warning css-dev-only-do-not-override-1uq9j6g"
            >
              1000
            </span>
          </div>
        </li>
      </div>
    </div>
  </div>
</div>
`;
